### PR TITLE
migrate NuGetGallery.Facts to sdk csproj

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

File: tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj [hunk 1]
```diff
@@ -1,402 +1,50 @@
...
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <Prefer32Bit>false</Prefer32Bit>
+  <PropertyGroup>
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:5]: We put generally put this sign block at the end of the csproj, for consistency. It might be worth considering importing https://github.com/NuGet/NuGetGallery/blob/main/SdkProjects.props like the other existing SDK projects. Given these are test projects some of the properties might not be needed, so maybe we could toggle the non-test properties with "is not test project" like condition.

Comment (SimonCropp) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj]: moved

> It might be worth considering importing

i will do that as a single PR across all test projects once they are all on the new sdk

File: tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj [hunk 2]
```diff
@@ -1,402 +1,50 @@
...
+    <ProjectReference Include="..\..\src\NuGetGallery.Services\NuGetGallery.Services.csproj" />
+    <ProjectReference Include="..\..\src\NuGetGallery\NuGetGallery.csproj" />
+    <ProjectReference Include="..\NuGetGallery.Core.Facts\NuGetGallery.Core.Facts.csproj" />
+
+    <PackageReference Include="Microsoft.Data.Services" Version="5.8.4" />
+    <PackageReference Include="Moq" Version="4.8.2" />
+    <PackageReference Include="xunit.abstractions" Version="2.0.3" />
+    <PackageReference Include="xunit.assert" Version="2.5.0" />
+    <PackageReference Include="xunit.core" Version="2.3.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.0" PrivateAssets="all" />
```
Comment (erdembayar) [inline tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:43]: Keep the `<IncludeAssets>runtime; build; native; contentfiles</IncludeAssets>`?

Comment (SimonCropp) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:43]: r u sure? ` PrivateAssets="all"` is equivalent to `<IncludeAssets>runtime; build; native; contentfiles</IncludeAssets>` in this context. and less code

Comment (erdembayar) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:43]: See `all` here it includes more assets such as `compile, buildTransitive` etc: https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets

File: tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj [hunk 3]
```diff
@@ -1,402 +1,50 @@
...
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <Prefer32Bit>false</Prefer32Bit>
+    <TargetFramework>net472</TargetFramework>
   </PropertyGroup>
+  <Import Project="$(SignPath)\sign.targets" Condition="Exists('$(SignPath)\sign.targets')" />
   <ItemGroup>
-    <Reference Include="System" />
     <Reference Include="System.ComponentModel.Composition" />
```
Comment (erdembayar) [inline tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:7]: Could we remove all those remaining `System` and `Microsoft` references?

Comment (SimonCropp) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj]: done. BTW you know u can suggest changes inline in a GH PR
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

File: tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj [hunk 4]
```diff
@@ -1,402 +1,39 @@
...
+    <ProjectReference Include="..\..\src\NuGetGallery\NuGetGallery.csproj" />
+    <ProjectReference Include="..\NuGetGallery.Core.Facts\NuGetGallery.Core.Facts.csproj" />
+
+    <PackageReference Include="Microsoft.Data.Services" Version="5.8.4" />
+    <PackageReference Include="Moq" Version="4.8.2" />
+    <PackageReference Include="xunit.abstractions" Version="2.0.3" />
+    <PackageReference Include="xunit.assert" Version="2.5.0" />
+    <PackageReference Include="xunit.core" Version="2.3.1" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.0">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
```
Comment (erdembayar) [inline tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:30]: <ExcludeAssets>analyzers</ExcludeAssets> is missing

Comment (SimonCropp) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:30]: `analyzers</IncludeAssets>` ?

Comment (erdembayar) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:30]: Please pay attention to `IncludeAssets and ExcludeAssets`, we fixed same problems in other PRs too.
image

Comment (SimonCropp) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:30]: why are analyzers being excluded?

this is the recomended 

``

Comment (erdembayar) [reply tests/NuGetGallery.Facts/NuGetGallery.Facts.csproj:30]: It predates me. Most likely it doesn't matter since it's a test project, but I don't want to break an unknown edge case in our CI pipeline that could be set by this. Just copying existing asset values is very easy, so let's just keep it.

**Comment by erdembayar** [comment]
@SimonCropp 
Could you please address my comment above? If you are unable to work on this anytime soon, please close the and reopen it later when it's ready.
